vnext/Microsoft.ReactNative.Cxx/JSValueTreeWriter.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#ifndef MICROSOFT_REACTNATIVE_JSVALUETREEWRITER
+#define MICROSOFT_REACTNATIVE_JSVALUETREEWRITER
+
+#include <stack>
+#include "JSValue.h"
+
+namespace winrt::Microsoft::ReactNative::Bridge {
+
+// Writes to a tree of JSValue objects.
+struct JSValueTreeWriter : implements<JSValueTreeWriter, IJSValueWriter> {
```

Comment:
This is a bit general, but what are the main benefits we see of this compared to using Dynamic and DynamicConverter? Eithout context, it feels like there's some overlap.

Reply:
folly::dynamic is not ABI safe. We cannot use it in the API. But developers want to have something like folly-dynamic is some cases. The JSValue is an immutable class to fulfill that need. We could use the folly::dynamic, but I want the native modules code that is outside of ABI boundary to not have any dependencies on any big third-party libraries. The Microsoft.ReactNative.Cxx is a small shared code library that we require to depend on.

---
In reply to: [356918350](https://github.com/microsoft/react-native-windows/pull/3760#discussion_r356918350) [](ancestors = 356918350)